### PR TITLE
Logging `pip` download events

### PR DESCRIPTION
**Pull request by dorfire**

### CODE REVIEW

File: src/BuildScriptGeneratorCli/ITextStreamProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+namespace Microsoft.Oryx.BuildScriptGenerator
+{
+    public interface ITextStreamProcessor
```
Comment (gpcastro) [inline src/BuildScriptGeneratorCli/ITextStreamProcessor.cs]: does it have to be public? can we make it internal?

File: src/BuildScriptGeneratorCli/PipDownloadEventLogger.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Oryx.Common;
+
+namespace Microsoft.Oryx.BuildScriptGenerator
+{
+    internal class PipDownloadEventLogger : ITextStreamProcessor
```
Comment (gpcastro) [inline src/BuildScriptGeneratorCli/PipDownloadEventLogger.cs]: add some description for this is supposed to do

File: src/BuildScriptGeneratorCli/PipDownloadEventLogger.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        public PipDownloadEventLogger(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public void ProcessLine(string line)
+        {
+            if (_currentDownload != null)
+            {
+                _currentDownload.AddProperty("nextLine", line.Trim());
```
Comment (gpcastro) [inline src/BuildScriptGeneratorCli/PipDownloadEventLogger.cs]: can `line` contain PII?

File: src/BuildScriptGeneratorCli/PipDownloadEventLogger.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                _currentDownload.Dispose();
+                _currentDownload = null;
+            }
+
+            if (line.Contains(PipDownloadMarker))
```
Comment (gpcastro) [inline src/BuildScriptGeneratorCli/PipDownloadEventLogger.cs]: "Downloading" seems too broad... isn't there a regex that we can use instead?

Comment (dorfire) [reply src/BuildScriptGeneratorCli/PipDownloadEventLogger.cs]: This seems to be enough - especially considering *this checker only runs for Python builds*.
We could come up with a regex, but my guess is that it'll be difficult to write and to maintain

**Pull request merged**